### ISSUE TITLE
Limit login to groups

### ISSUE THREAD

**Issue by kousu** [zorn-v/nextcloud-social-login#396]
In my team we have a general #announcements channel in Discord for our followers, and then most of the work is kept in private channels behind a `Coordinators` role. I've been doing https://github.com/zorn-v/nextcloud-social-login/issues/390 because I want to map `Coordinators` on Discord to `Coordinators` on NextCloud. I've used the "Allow login only for specified guilds" to limit the blast radius, but it would be even better if I could also "Allow login only for the specified roles".

So that's my feature request.

Maybe this isn't something that should be solved in your plugin. Maybe we should reserve Discord for our core members and find a different medium for announcements. But if I could solve it here that would simplify my life.

**Comment by zorn-v** [comment]
> Maybe this isn't something that should be solved in your plugin

It is. I can't invent how to resolve your edge case in general way.

**Comment by kousu** [comment]
I just noticed

> * [ ] Restrict login for users without mapped groups
> * [ ] Restrict login for users without assigned groups

![Screenshot 2023-02-15 at 14-42-36 Social login - Administration settings - Yielding Opportunities for Terpsichore's Art](https://user-images.githubusercontent.com/987487/219135996-a590b193-0b52-4fc4-b552-ea2534533b8d.png)

These should solve my problem, once I get #390 finished. I overlooked them because at the moment I can't use them because there's no way to set up group mappings for Discord (or GitHub, Slack, Telegram, etc) -- basically only generic OIDC and Discourse can do it.

Sorry for the noise, and thanks for the very useful plugin.
